### ISSUE TITLE
Breakage in 1.7+ with QR factorization

### ISSUE THREAD

**Issue by gbaraldi** [JuliaArrays/StaticArrays.jl#931]
Calling qr on a simple SMatrix is now giving ambiguous method error on versions after 1.6.1
1.6.1
```julia
using LinearAlgebra, StaticArrays; qr(SA[1 2; 3 4])

StaticArrays.QR{SMatrix{2, 2, Float64, 4}, SMatrix{2, 2, Float64, 4}, SVector{2, Int64}}([-0.316227766016838 -0.9486832980505138; -0.9486832980505138 0.316227766016838], [-3.1622776601683795 -4.427188724235731; 0.0 -0.6324555320336751], [1, 2])
```
On 1.7-beta2
```julia
julia> using LinearAlgebra, StaticArrays; qr(SA[1 2; 3 4])
ERROR: MethodError: qr(::SMatrix{2, 2, Int64, 4}, ::Val{false}) is ambiguous. Candidates:

qr(A::AbstractMatrix, ::Val{false}; kwargs...) in LinearAlgebra at deprecated.jl:70
  qr(A::StaticMatrix, pivot::Union{Val{false}, Val{true}}) in StaticArrays at /Users/gabrielbaraldi/.julia/packages/StaticArrays/aMe0r/src/qr.jl:37
Possible fix, define
  qr(::StaticMatrix, ::Val{false})
Stacktrace:
 [1] qr(A::SMatrix{2, 2, Int64, 4})
   @ StaticArrays ~/.julia/packages/StaticArrays/aMe0r/src/qr.jl:38
 [2] top-level scope
   @ REPL[3]:1
```

On 1.8
```julia
julia> using LinearAlgebra, StaticArrays; qr(SA[1 2; 3 4])
ERROR: MethodError: qr(::SMatrix{2, 2, Int64, 4}, ::Val{false}) is ambiguous. Candidates:
  qr(A::AbstractMatrix, ::Val{false}; kwargs...) in LinearAlgebra at deprecated.jl:70
  qr(A::StaticMatrix, pivot::Union{Val{false}, Val{true}}) in StaticArrays at /Users/gabrielbaraldi/.julia/packages/StaticArrays/aMe0r/src/qr.jl:37
Possible fix, define
  qr(::StaticMatrix, ::Val{false})
```
This looks like a julia issue, but I'm not sure, since it affects at least StaticArrays I am opening an issue here and an issue upstream.

**Comment by gbaraldi** [comment]
It seems to have been introduced by https://github.com/JuliaLang/julia/pull/40623

**Comment by mateuszbaran** [comment]
This looks similar to https://github.com/JuliaArrays/StaticArrays.jl/issues/920 and very likely requires a similar fix.

**Comment by mateuszbaran** [comment]
This is now fixed by #932
